### PR TITLE
Merged area hook/context PR into edit PR

### PR DESCRIPTION
**Pull request by paterasMSFT**

### CODE REVIEW

File: src/AreaView/AreaCard/Details/AreaCardDetailsEdit.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import { useStateValue, IOKRContext } from "../../../StateMangement/StateProvider";
+
+export const AreaCardDetailsEdit: React.FunctionComponent = props => {
+	const stateContext = useStateValue();
+	const [{ area }, areaDispatcher] = useAreaCardValue();
+
+	return (
+		<div className="card-header">
+			<h3>
+				<MutableField value={area.Name} onChange={(val) => { area.Name = val; }} />
```
Comment (beccamc) [inline src/AreaView/AreaCard/Details/AreaCardDetailsEdit.tsx]: What state value are you updating here? How does the cancel work?

Comment (beccamc) [reply src/AreaView/AreaCard/Details/AreaCardDetailsEdit.tsx]: If this is directly updating the area value from useAreaCardValue, that's the "global" state, (only for this dispatcher, but still "global"). When the reducer is fired to set the area = action.area, isn't that already set?

Comment (paterasMSFT) [reply src/AreaView/AreaCard/Details/AreaCardDetailsEdit.tsx]: Forgot to do Name. Good catch.

File: src/AreaView/AreaCard/Details/AreaCardDetailsEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { MutableField } from "../../../MutableField";
```
[the remark this answers is not in the document]
Comment (paterasMSFT) [reply src/AreaView/AreaCard/Details/AreaCardDetailsEdit.tsx]: Yeah, I was thinking about changing that. You have to click twice to submit it, too. I'll switch to inputs.

File: src/AreaView/AreaCard/Details/AreaCardDetailsEdit.tsx (hunk 1 above)
Comment (paterasMSFT) [reply src/AreaView/AreaCard/Details/AreaCardDetailsEdit.tsx]: As for cancel, I was getting it for free from MutableField (it only fires onChange when it loses focus, and it discards on escape).

**Pull request merged**